File: backend/src/main/java/com/example/meal_catalogue_planner/dto/MealPlanResponse.java
package com.example.meal_catalogue_planner.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

import com.example.meal_catalogue_planner.entity.MealType;

// Response body returned for meal plan reads and writes.
public record MealPlanResponse (
    // Database ID of the meal plan.
    Long id,

    // User-facing name for the meal plan.
    String name,

    // Calendar date assigned to the meal plan.
    LocalDate mealDate,

    // Meal slot, such as breakfast, lunch, or dinner.
    MealType mealType,

    // Optional free-form notes for the meal plan.
    String notes,

    // Food item rows included in the meal plan.
    List<MealPlanItemResponse> items,

    // Total calories calculated from all item row subtotals.
    BigDecimal totalCalories
) {
    
}

File: frontend/src/app/models/meal-plan-response.ts
import { MealPlanItemResponse } from "./meal-plan-item-response"
import { MealType } from "./meal-type"

// Meal plan data returned by the backend.
export interface MealPlanResponse {
    // Database ID of the meal plan.
    id: number
    // User-facing meal plan name.
    name: string
    // Calendar date in backend-compatible string form.
    mealDate: string
    // Meal slot for the plan.
    mealType: MealType
    // Optional notes entered by the user.
    notes: string
    // Food rows included in the meal plan.
    items: MealPlanItemResponse[]
    // Total calories across all meal plan item rows.
    totalCalories: number
}
